DESIGN.md

```diff
@@ -103,8 +93,14 @@ ram_per_jruby = (512, 768, 1024) if total memory (4-7 GB, 8-16 GB, 16 GB+)
 ram_per_jruby_code_cache = 128
 ```
 
-PuppetServer jrubies are constrained based on both how many jrubies fit into unallocated memory and unallocated processors. PuppetServer memory is then set the amount of memory required for all jrubies.
+PuppetServer jrubies are constrained based on both how many jrubies fit into unallocated memory and unallocated processors (one per processor). PuppetServer memory is then set the amount of memory required for all jrubies.
 
+```
```

Comment:
Add
```
MAX_JRUBIES = 24 # can we provide the reason for this cap?
```

Reply:
changed 24 to maximum_cpu_puppetserver  as it is described above (along with why 24 is the cap)

Reply:
Detail: support has seen this: `CodeCache is full. Compiler has been disabled` probably because `UseCodeCacheFlushing` defaults to false ...

UseCodeCacheFlushing: Attempt to sweep the codecache before shutting off compiler